app/controllers/albums.js

```diff
@@ -1,13 +1,46 @@
 const logger = require('../logger'),
-  bcrypt = require('bcryptjs'),
   albumService = require('../services/albums'),
-  User = require('../models').users,
-  userMw = require('../middlewares/users'),
   errors = require('../errors'),
-  fetch = require('node-fetch');
+  Album = require('../models').albums_bought;
 
 exports.getAll = (request, response, next) => {
   if (!request.userLogged)
     return response.status(400).send(errors.authenticationError(['You must be logged to see all albums']));
   albumService.getAll(response).then(data => response.status(200).send(data));
 };
+
+exports.buy = (request, response) => {
+  if (!request.userLogged)
+    return response.status(400).send(errors.authenticationError('You must be logged to buy albums'));
+  const handleError = (err, message) => {
+    logger.error(err);
+    response.status(400).send(errors.buyAlbumError(message));
+  };
+  const id = Number(request.params.id);
+  albumService
+    .getAll(response)
```

Comment:
I don't remember if we talked about this, but the album service api supports a search by id. You can do something like `https://jsonplaceholder.typicode.com/albums/albumId` and it will respond only with the album matching that id. That way you can avoid asking for the whole list of albums all the time.